plugin/ochttp/client.go

```diff
@@ -43,6 +43,11 @@ type Transport struct {
 	// for spans started by this transport.
 	StartOptions trace.StartOptions
 
+	// NameFromRequest holds the function to use for generating the span name
+	// from the information found in the outgoing HTTP Request. By default the
+	// name equals the URL Path.
+	NameFromRequest func(*http.Request) string
```

Comment:
Could we have a unit test for this new functionality?

Reply:
sure